ocp_utilities/monitoring.py

```diff
@@ -0,0 +1,137 @@
+from json import JSONDecodeError
+import requests
+
+from ocp_resources.route import Route
+from ocp_resources.secret import Secret
+from ocp_resources.service_account import ServiceAccount
+from ocp_resources.utils import TimeoutExpiredError, TimeoutSampler
+from ocp_utilities.infra import get_client
+
+TIMEOUT_10MIN = 10 * 60
+
+LOGGER = logging.getLogger(__name__)
+
+class Prometheus(object):
+    """
+    For accessing Prometheus cluster metrics
+
+    Prometheus HTTP API doc:
+    https://prometheus.io/docs/prometheus/latest/querying/api/
+
+    Argument for query method should be the entire string following the server address
+        e.g.
+        prometheus = Prometheus()
+        up = prometheus.query("/api/v1/query?query=up")
+    """
+
+    def __init__(
+        self,
+        namespace="openshift-monitoring",
+        resource_name="prometheus-k8s",
+        client=None,
+    ):
+        self.namespace = namespace
+        self.resource_name = resource_name
+        self.client = get_client()
+        self.api_v1 = "/api/v1"
+
+        # get route to prometheus HTTP api
+        self.api_url = self._get_route()
+
+        # get prometheus ServiceAccount token
+        self.headers = self._get_headers()
+
+    def _get_route(self):
+        # get route to prometheus HTTP api
+        LOGGER.info("Prometheus: Obtaining route")
+        route = Route(
+            namespace=self.namespace, name=self.resource_name, client=self.client
+        ).instance.spec.host
+
+        return f"https://{route}"
+
+    def _get_headers(self):
+        """Uses the Prometheus serviceaccount to get an access token for OAuth"""
+        LOGGER.info("Prometheus: Setting headers")
+
+        LOGGER.info("Prometheus headers: Obtaining OAuth token")
+
+        # get SA
+        prometheus_sa = ServiceAccount(
+            namespace=self.namespace, name=self.resource_name, client=self.client
+        )
+
+        # get secret
+        secret_name = prometheus_sa.instance.imagePullSecrets[0].name
+        secret = Secret(namespace=self.namespace, name=secret_name, client=self.client)
+
+        # get token value
+        token = secret.instance.metadata.annotations["openshift.io/token-secret.value"]
+
+        return {"Authorization": f"Bearer {token}"}
+
+    def _get_response(self, query):
+        requests.packages.urllib3.disable_warnings()
+        response = requests.get(
+            f"{self.api_url}{query}", headers=self.headers, verify=False
+        )
+
+        try:
+            return json.loads(response.content)
+        except JSONDecodeError as json_exception:
+            LOGGER.error(
+                "Exception converting query response to JSON: "
+                f"exc={json_exception} response_status_code={response.status_code} response={response.content}"
+            )
+            raise
+
+    def query(self, query):
+        return self._get_response(query=f"{self.api_v1}/query?query={query}")
+
+    def get_alert(self, alert):
```

Comment:
this should utilize self.alerts() and filter the one by names. Please note: same alert name against multiple components may exists on a given cluster, so it should return the list of all matched alerts.

Reply:
@dbasunag 
self.alerts() will always return the alerts that are currently firing on the cluster, whereas query method for getting response based on specific query, your suggestion could be used for use case based test implementation but for util I couldn't see if that is useful or not
If you can please elaborate more or provide any suitable use case it would be more helpful

Reply:
Using query endpoint to find alerts in problematic. As it would show you all the alerts that were fired within default time period (5m or 30m), including alerts that has since been resolved. If someone wants to make queries to alerts, they should use query() call explicitly. Alert call should give us current active alerts (using alerts()). self.alerts() returns not just firing alerts, also all the active alerts (including pending one). We should be mimicking:  https://prometheus.io/docs/prometheus/latest/querying/api/#alerts

Reply:
@dbasunag 
Ok, I'll update get_alert to return alerts matching with self.alerts
currently get_alert query the alert based on the input alert provided . i'll update it return the response if it matches with self.alerts otherwise it will return alert not active

Reply:
pushed the changes as requested, please verify

Reply:
@dbasunag 
updated the get_alert function also alert sample based on the new alert, please verify